.github/workflows/build-and-test.yml

```
@@ -87,15 +89,19 @@ jobs:
           text: 'Build failed :sadpanda:'
           username: 'Java-PFB GitHub Action'
 
-  dispatch-tag:
+  tag:
     needs: [ build, unit-tests-and-sonarqube, source-clear ]
-    runs-on: ubuntu-latest
+    uses: ./.github/workflows/tag.yml
+    if: success() && github.ref == 'refs/heads/main'
 
+  publish-library:
+    needs: [ build, unit-tests-and-sonarqube, source-clear, tag ]
+    uses: ./.github/workflows/publish.yml
     if: success() && github.ref == 'refs/heads/main'
 
-    steps:
-      - name: Fire off tag action
-        uses: broadinstitute/workflow-dispatch@v1
-        with:
-          workflow: Tag
-          token: ${{ secrets.BROADBOT_TOKEN }}
+  release-cli:
+    needs: [ build, unit-tests-and-sonarqube, source-clear, tag, publish-library ]
```

same change here, don't need to include transitive list of workflow jobs
```suggestion
    needs: [ publish-library ]
```